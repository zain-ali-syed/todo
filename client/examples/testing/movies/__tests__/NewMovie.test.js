import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import NewMovie from '../NewMovie';


afterEach(cleanup);

test('<NewMovie />', (useSnapshot = true) => {
  console.log('useSnapshot ', useSnapshot);
  const {
    debug, getByTestId, queryByTestId, container,
  } = render(<NewMovie />);
  // debug();

  expect(getByTestId('title').textContent).toBe('New Movie');
  expect(queryByTestId('form-new-movie')).toBeTruthy();
  if (useSnapshot) expect(container.firstChild).toMatchSnapshot();
});
