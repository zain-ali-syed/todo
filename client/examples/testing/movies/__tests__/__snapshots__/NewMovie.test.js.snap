// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie /> 1`] = `
<div>
  <h3
    data-testid="title"
  >
    New Movie
  </h3>
  <div>
    <form
      data-testid="form-new-movie"
    >
      <label
        for="title"
      >
        Title
        <input
          type="text"
          id="title"
          name="title"
        />
      </label>
      <button
        type="submit"
        data-testid="submit_new_movie"
      >
        Add Movie
      </button>
    </form>
  </div>
</div>
`;
